cpp/include/cudf/wrappers/dictionary.hpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+#include <cuda_runtime.h>
+
+/**
+ * @file dictionary.hpp
+ * @brief Concrete type definition for dictionary columns.
+ */
+namespace cudf
+{
+
+/**
+ * @brief A strongly typed wrapper for indices in a DICTIONARY type column.
+ *
+ * IndicesType will be integer types like int32_t.
```

Comment:
I think this should be `IndexType`, but it's not a huge deal. (I think you probably would say `KeyType`, not `KeysType`, so logically should use singular Index.)